router/core/websocket.go

```diff
@@ -308,6 +311,35 @@ func (h *WebsocketHandler) handleUpgradeRequest(w http.ResponseWriter, r *http.R
 		return
 	}
 
+	// Authenticate the connection using the initial payload
+	fromInitialPayloadConfig := h.config.Authentication.FromInitialPayload
+	if fromInitialPayloadConfig.Enabled {
+		// Setting the initialPayload in the context to be used by the websocketInitialPayloadAuthenticator
+		r = r.WithContext(authentication.WithWebsocketInitialPayloadContextKey(r.Context(), handler.initialPayload))
+
+		// Later check access control after initial payload is read and set into the context
+		validatedReq, err := h.accessController.Access(w, r)
+		if err != nil {
+			statusCode := http.StatusForbidden
+			if errors.Is(err, ErrUnauthorized) {
+				statusCode = http.StatusUnauthorized
+			}
+			http.Error(handler.w, http.StatusText(statusCode), statusCode)
+			handler.writeErrorMessage(requestID, err)
+			handler.Close()
+			return
+		}
+		handler.r = validatedReq
+
+		// Export the token from the initial payload to the request header
+		if fromInitialPayloadConfig.ExportToken.Enabled {
+			var initialPayloadMap map[string]interface{}
+			json.Unmarshal(handler.initialPayload, &initialPayloadMap)
```

Comment:
The previous discussion was closed without a comment. Why is the err not handled?